Bottom: 3364fe6dd47894ba374351c71bd944e8d8b76eae
Top:    a32d5ba6ef37ea4968671f42c0343a7799bce16f
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-08-18 16:35:02 +0800

vhost_net: use __skb_array_len_with_tag() in peek_head_len()

Reuse instead of open code __skb_array_len_with_tag().

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index ba08b78..37cb77f 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -583,7 +583,6 @@ static void handle_tx(struct vhost_net *net)
 
 static int peek_head_len(struct vhost_net_virtqueue *rvq, struct sock *sk)
 {
-	struct sk_buff *head;
 	int len = 0;
 	unsigned long flags;
 
@@ -591,13 +590,7 @@ static int peek_head_len(struct vhost_net_virtqueue *rvq, struct sock *sk)
 		return vhost_net_buf_peek(rvq);
 
 	spin_lock_irqsave(&sk->sk_receive_queue.lock, flags);
-	head = skb_peek(&sk->sk_receive_queue);
-	if (likely(head)) {
-		len = head->len;
-		if (skb_vlan_tag_present(head))
-			len += VLAN_HLEN;
-	}
-
+	len = __skb_array_len_with_tag(skb_peek(&sk->sk_receive_queue));
 	spin_unlock_irqrestore(&sk->sk_receive_queue.lock, flags);
 	return len;
 }
